api/nfd/v1alpha1/types.go

```diff
@@ -131,6 +131,54 @@ type NodeFeatureRuleSpec struct {
 	Rules []Rule `json:"rules"`
 }
 
+// NodeFeatureGroup resource holds Node pools by featureGroup
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=nfg
```

Comment:
Hmm, I think we should handle status as a separate subresource

```suggestion
// +kubebuilder:resource:scope=Namespaced,shortName=nfg
// +kubebuilder:subresource:status
```

Then nfd-master will only update the status subresource and not touch the other parts

Reply:
You changed this, right? Can be closed as resolved(?)